frontend/src/metabase/actions/containers/ActionCreator/FormCreator/OptionEditor.tsx

```diff
@@ -16,22 +16,36 @@ import {
 } from "./OptionEditor.styled";
 
 const optionsToText = (options: FieldValueOptions) => options.join("\n");
-const textToOptions = (text: string): FieldValueOptions =>
-  text.split("\n").map(option => option.trim());
+const textToOptions = (text: string): FieldValueOptions => {
+  const options = text
+    .trim()
+    .split("\n")
+    .map(option => option.trim())
+    .filter(Boolean);
+  const uniqueOptions = [...new Set(options)];
 
-function cleanOptions(options: FieldValueOptions, fieldType: FieldType) {
+  return uniqueOptions;
+};
+
+function transformOptionsIfNeeded(
+  options: FieldValueOptions,
+  fieldType: FieldType,
+) {
   if (fieldType === "number") {
     return options.map(option => Number(option));
   }
+
   return options;
 }
 
 function getValidationError(options: FieldValueOptions, fieldType: FieldType) {
-  if (fieldType === "number") {
-    const isValid = options.every(option => !Number.isNaN(option));
-    return isValid ? undefined : t`Invalid number format`;
+  if (fieldType !== "number") {
+    return;
   }
-  return;
+
+  const isValid = options.every(option => !Number.isNaN(option));
```

Comment:
I know this code is just moved but it got my attention.
I believe the original intention was to check `isValid = options.every(isNumber)`, but:

```js
getValidationError([5], "number") // undefined - as expected
getValidationError([5, "5", "definitely-not-a-number", Symbol(), {}, []], 'number') // undefined - not expected
```

I suggest to fix it and add tests for that function (also extract it to utils).

Reply:
@kamilmielnik Another thing going on here is that we're using `Number.isNaN` vs `isNaN` which seem to be very different functions despite the similar names (https://stackoverflow.com/questions/33164725/confusion-between-isnan-and-number-isnan-in-javascript)

For example:
<img width="297" alt="image" src="https://github.com/metabase/metabase/assets/25306947/6afe16aa-0ea0-438e-9d85-9d41a72b1935">

Reply:
@kamilmielnik you're right, but if you check `transformOptionsIfNeeded` method which we call just before `getValidationError`, you'll see that we converted all the options to Number type, so values are either numbers or `NaN`

so it's more or less correct 
<img width="271" alt="image" src="https://github.com/metabase/metabase/assets/125459446/f6ff85e8-dbe6-4d14-bcc5-b5b7a83d8596">

Reply:
two methods could be merged together for sure